src/map/ai/states/mobskill_state.cpp

```diff
@@ -84,11 +84,22 @@ CMobSkill* CMobSkillState::GetSkill()
 
 void CMobSkillState::SpendCost()
 {
+    auto tp = 0;
     if (!m_PSkill->isTpFreeSkill())
     {
-        m_spentTP            = m_PEntity->health.tp;
-        m_PEntity->health.tp = 0;
-    }
+        if (m_PEntity->StatusEffectContainer->HasStatusEffect(EFFECT_SEKKANOKI))
+        {
+            tp = m_PEntity->addTP(-1000);
```

Comment:
why define a new variable when m_spentTP already exists?

Reply:
ahh I see what you mean now.